### PR TITLE
Add LDAP Plugin Support

### PR DESCRIPTION
**Pull request by zicklag**
Add support for LDAP which can be configured through environment
variables.

### CODE REVIEW

File: Dockerfile
Comment (zicklag) [inline Dockerfile:37]: I'm not sure if you want to add all of the new environment variables to the Dockerfile or not, but they all have sensible defaults in the `docker-settings.py` file, so it will work either way.

File: README.md
Comment (zicklag) [inline README.md:90]: I added a troubleshooting section if you don't mind it being in the same PR.

**Review by zicklag — COMMENTED** [review]
Here are a couple of questions.

**Comment by zicklag** [comment]
It's completely up to you, but it could be useful to mention my company's Docker ApacheDS LDAP server [image](https://github.com/katharostech/docker_apacheds) as on option for hosting LDAP in Docker.

I can add it to the README if you think it is a good idea.

**Comment by blackandred** [comment]
Sorry, I didn't notice the PR before. I will check it in few hours.

File: Dockerfile
Comment (blackandred) [inline Dockerfile:102]: Is this a this temporary workaround or a final solution for the OpenSSL issue?

File: README.md
Comment (blackandred) [reply README.md:90]: Very, very good :wink:

File: Dockerfile
Comment (blackandred) [reply Dockerfile:37]: I think that we should put everything we support into the Dockerfile for clarity.
Please add it in format:

```
    # what this does 
    SOME_ENV=defaultvalue \
    # what this does, a comment
    SOME_OTHER_ENV=some
```

As I will port a dynamic documentation generation based on Dockerfile content like it is there: https://github.com/riotkit-org/smtp-ext-relay/blob/master/README.md.j2#L54

**Review by blackandred — CHANGES_REQUESTED** [review]
Please add the env variables to the Dockerfile :wink:

File: Dockerfile
Comment (zicklag) [reply Dockerfile:102]: I think it's probably a final solution. It just relaxes the security requirements for the ciphers that it will allow connections over. If we wanted to we could leave it out and it would force people to upgrade the cipher suite on their LDAP servers, but I don't think it is our place to do so. If they want to make their LDAP server more secure that is up to them.

If we left it out, it wouldn't technically be a "problem" so to speak with the container, it would just refuse to connect to any LDAP servers that aren't at certain minimum security level. I think, though, that it makes sense to do it like this.

Comment (blackandred) [reply Dockerfile:102]: Ok, sure.

**Comment by zicklag** [comment]
> As I will port a dynamic documentation generation based on Dockerfile content like it is there: https://github.com/riotkit-org/smtp-ext-relay/blob/master/README.md.j2#L54

Ooh, that's cool. :+1:

---

I should have the Dockerfile updated soon.

**Comment by blackandred** [comment]
> > As I will port a dynamic documentation generation based on Dockerfile content like it is there: https://github.com/riotkit-org/smtp-ext-relay/blob/master/README.md.j2#L54
> 
> Ooh, that's cool. 
> 
> I should have the Dockerfile updated soon.

Cool :+1: 
I will create PR with README automation and build version automation in a few days, after your will be merged.

**Comment by zicklag** [comment]
OK, it should be ready now, but I haven't build and tested it yet. I'll let you know when I've tested it.

**Comment by blackandred** [comment]
Sure, I will wait. It would be good to setup a CI for pull requests :thinking:

**Comment by zicklag** [comment]
If you don't mind [Drone](https://drone.io) I could setup automated builds using that. You can't run containers on Drone, but you can build them using [Kaniko](https://github.com/GoogleContainerTools/kaniko). I do that with my containers, like [this one](https://github.com/katharostech/docker_dokuwiki/blob/master/.drone.yml).

If you want to use Travis, that makes sense so don't worry about it, but I figured I'd bring it up just in case.

**Comment by blackandred** [comment]
Sure, you could setup whatever if you can maintain it :) I will set up
travis as I do in other projects, I like everything to be done in the same
way :)

**Comment by zicklag** [comment]
Go ahead and do Travis. :+1: :slightly_smiling_face:

File: Dockerfile
Comment (blackandred) [reply Dockerfile:37]: Reminder: @zicklag I'm waiting for the env change to be applied, then I will merge the PR.

Comment (zicklag) [reply Dockerfile:37]: Oh, I meant to resolve this conversation. The env change is already in the PR.

**Review by blackandred — APPROVED** [review]

**Pull request merged**
